### DISCUSSION TITLE
OpenAPI servers not being returned according how the docs say they should be

### DISCUSSION THREAD

**Discussion by mzealey** [fastapi/fastapi#12226] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

# you can add a test endpoint here or not - same bug either way
```

### Description

```
$ curl localhost:8$ curl localhost:8000/openapi.json
{"openapi":"3.1.0","info":{"title":"FastAPI","version":"0.1.0"},"paths":{}}
```

According to the documentation of the `servers` parameter in FastAPI:

> If the servers list is not provided, or is an empty list, the default value would be a dict with a url value of /.

(assuming that `root_path_in_servers = True` (the default))

Clearly this is not happening.

### Operating System

Linux

### FastAPI Version

0.110.3 (but according to github code seems to be in latest also)

### Pydantic Version

2.5.3

### Python Version

Python 3.10.12

**Comment by sinisaos** [comment]
@mzealey I think the problem is in this line. 

https://github.com/fastapi/fastapi/blob/7c6f2f8fde68f488163376c9e92a59d46c491298/fastapi/openapi/utils.py#L484-L485

If we change it with something like this

```python
output["servers"] = [] if servers is None else servers
```

the result is this which I think is correct. 

```bash
> curl localhost:8000/openapi.json
{"openapi":"3.1.0","info":{"title":"FastAPI","version":"0.1.0"},"servers":[],"paths":{}}
```

Sorry if I miss your point.

**Reply by mzealey** [reply]
My point is not that the behaviour is necessarily wrong, it's that it is quite different from the documentation. My reading of the documentation (as posted in my first question) is that the `servers` not specified, or if it is `[]` or `None` then the output should be equal to:

```
"servers":[{"url": "/"}]
```

(assuming that `root_path_in_servers = True` (the default))

So either the documentation needs updating, or (ideally) the openapi output does.

**Comment by Kludex** [comment]
Can you point me to the documentation section?

**Reply by mzealey** [reply]
https://github.com/fastapi/fastapi/blob/9606b916ef7c883f5ebeac1bf3db9adf5ae646a9/fastapi/applications.py#L299

**Reply by Kludex** [reply, marked as answer]
Created an issue: https://github.com/fastapi/fastapi/issues/12246.

Thanks.
